driver/compile.go

```diff
@@ -181,6 +182,189 @@ func setGroupByProcInputSortDir(p ast.Proc, inputSortField string, inputSortDir
 	}
 }
 
+// expressionFields returns a slice with all fields referenced
+// in an expression. Fields will be repeated if they appear
+// repeatedly.
+func expressionFields(expr ast.Expression) []string {
```

Comment:
Nits: This and booleanExpressionFields would be a little lighter if we changed the parameter to e. Also, maybe s/expression/expr/ in function names; the abbreviation is well-known and would render the names more wieldy.